Application/src/controller/network/Net.java

```diff
@@ -102,8 +102,15 @@ private void inner_loop() {
                     // Run the decode on the Platform instance.
                     Platform.runLater(() -> {
 
+                        // Pull a message from the received messages queue.
```

Comment:
Why do you have changes in this file?

Reply:
I found a bug which was occurring when you try to open multiple diagrams at once i.e. a sequence and it's parallel at the same time. It would run out of items in the queue while still pulling.

Reply:
So it's very related to my feature.